lib/spack/env/cc

```diff
@@ -98,25 +98,25 @@ case "$command" in
     cpp)
         mode=cpp
         ;;
-    cc|c89|c99|gcc|clang|icc|pgcc|xlc)
+    cc|c89|c99|gcc|clang|icc|pgcc|xlc|xlc_r)
```

Comment:
These flags are required by the XL Fortran compiler in order to be able to build test files provided by cmake that are used to test compiler functionality.  Spack users will need to use those flags in order to be able to build their packages and I don't want to keep on reminder users to add those flags.  So I want them added by default and the best place is in the compiler class.  The changes I suggested for compilers/__init__.py file take care of that.